pkg/storj/encryption.go

```diff
@@ -131,6 +131,14 @@ const (
 	NonceSize = 24
 )
 
+// NewKey creates a new key from a passphrase
+func NewKey(passphrase []byte) *Key {
+	// TODO: if/v3-1541 This function should return an error when len(passphrase) == 0
```

Comment:
I don't think we need such an error. The Key type has the IsZero method, so we should be valid to create a empty/zero key too.

Reply:
I added this after knowing that the current `cmd/uplink` doesn't work with a zero key value, because `lib/uplink` rejects it.

Reply:
Do we allow an empty passphrase?

Reply:
we should not allow an empty passphrase. if someone wants an empty passphrase really want they want is to disable encryption entirely. if we're dealing with passphrases they should not be empty

Reply:
We've been talking in other places about how "passphrase" needs to be differentiated from "key material". Users might want to have a high-entropy key and encrypt that with a passphrase, as opposed to deriving the key from the passphrase via some KDF. Or they might want a high-entropy key stored in the clear on their system because they trust their system security and want to have passphrase-less access. Or they might have a hardware token further confusing the situation.

The question about empty passphrases seems to contain an invalid assumption, given that.

Reply:
I'm not sure how to apply this to the scope of the issue.
The issue has the goal of only moving the encryption key from the `config.yaml` to a separated file nothing else.

`cmd/uplink` was already asking for a "passphrase", but that has a wrong name, because it's the encryption key itself.

My new PR #1967 has changed the `cmd/uplink` to also treat the "passphrase" for not being the key, it's used for generating a key through PBKDF2; hence the "passphrase" can be forgotten because it isn't used for anything else.

We want to go further with this topic, but I don't think that all belongs to this issue, because there is already another issue to have write a design doc for the key management.

Reply:
I hope that we start to see the scope of the issue, because otherwise, I suggest to leave this issue aside, write a design doc with all the workflow that they have for being agreed  and then do the implementation of it.